### PR TITLE
webflux client cleanup

### PR DESCRIPTION
**Pull request by richardstartin**
Rationale in inline comments

### CODE REVIEW

File: dd-java-agent/instrumentation/spring-webflux-5/src/main/java8/datadog/trace/instrumentation/springwebflux/client/WebClientTracingFilter.java [hunk 1]
```diff
@@ -50,24 +50,19 @@ public MonoWebClientTrace(final ClientRequest request, final ExchangeFunction ne
...
+        TraceWebClientSubscriber tracingSubscriber = new TraceWebClientSubscriber(subscriber, span);
         next.exchange(builder.build())
-            .doOnCancel(
-                () -> {
-                  DECORATE.onCancel(span);
-                  span.finish();
-                })
-            .subscribe(new TraceWebClientSubscriber(subscriber, span));
+            .doOnCancel(tracingSubscriber::onCancel)
+            .subscribe(tracingSubscriber);
```
Comment (richardstartin) [inline dd-java-agent/instrumentation/spring-webflux-5/src/main/java8/datadog/trace/instrumentation/springwebflux/client/WebClientTracingFilter.java:65]: No need to capture the span

File: dd-java-agent/instrumentation/spring-webflux-5/src/main/java8/datadog/trace/instrumentation/springwebflux/client/WebClientTracingFilter.java [hunk 2]
```diff
@@ -50,24 +50,19 @@ public MonoWebClientTrace(final ClientRequest request, final ExchangeFunction ne
 
     @Override
     public void subscribe(final CoreSubscriber<? super ClientResponse> subscriber) {
-      final AgentSpan span;
-      if (activeSpan() != null) {
-        span = startSpan(HTTP_REQUEST, activeSpan().context());
-      } else {
-        span = startSpan(HTTP_REQUEST);
-      }
+      AgentSpan activeSpan = activeSpan();
```
Comment (richardstartin) [inline dd-java-agent/instrumentation/spring-webflux-5/src/main/java8/datadog/trace/instrumentation/springwebflux/client/WebClientTracingFilter.java:53]: Don't look up the active span twice

File: dd-java-agent/instrumentation/spring-webflux-5/src/main/java8/datadog/trace/instrumentation/springwebflux/client/WebClientTracingFilter.java [hunk 3]
```diff
@@ -50,24 +49,15 @@ public MonoWebClientTrace(final ClientRequest request, final ExchangeFunction ne
 
     @Override
     public void subscribe(final CoreSubscriber<? super ClientResponse> subscriber) {
-      final AgentSpan span;
-      if (activeSpan() != null) {
-        span = startSpan(HTTP_REQUEST, activeSpan().context());
-      } else {
-        span = startSpan(HTTP_REQUEST);
-      }
+      AgentSpan span = startSpan(HTTP_REQUEST);
```
Comment (richardstartin) [inline dd-java-agent/instrumentation/spring-webflux-5/src/main/java8/datadog/trace/instrumentation/springwebflux/client/WebClientTracingFilter.java:52]: This reimplemented `startSpan`

File: dd-java-agent/instrumentation/spring-webflux-5/src/main/java8/datadog/trace/instrumentation/springwebflux/client/WebClientTracingFilter.java [hunk 4]
```diff
@@ -50,24 +49,15 @@ public MonoWebClientTrace(final ClientRequest request, final ExchangeFunction ne
...
+        TraceWebClientSubscriber tracingSubscriber = new TraceWebClientSubscriber(subscriber, span);
         next.exchange(builder.build())
-            .doOnCancel(
-                () -> {
-                  DECORATE.onCancel(span);
-                  span.finish();
-                })
-            .subscribe(new TraceWebClientSubscriber(subscriber, span));
+            .doOnCancel(tracingSubscriber::onCancel)
+            .subscribe(tracingSubscriber);
```
Comment (richardstartin) [inline dd-java-agent/instrumentation/spring-webflux-5/src/main/java8/datadog/trace/instrumentation/springwebflux/client/WebClientTracingFilter.java:60]: It's cleaner to concentrate the lifecycle management on same object

**Review by devinsba — APPROVED** [review]

**Review by mcculls — APPROVED** [review]

**Review by tylerbenson — APPROVED** [review]

**Pull request merged**